src/patch/Operation.ts

```diff
@@ -0,0 +1,202 @@
+import { isEqual } from '../utils';
+import JsonPointer, { createPointer } from './JsonPointer';
+export const enum OperationType {
+	Add,
+	Remove,
+	Replace,
+	Copy,
+	Move,
+	Test
+}
+
+interface Operation {
+	op: string;
+	path: JsonPointer;
+	toString(): string;
+	apply(target: any): any;
+}
+
+export default Operation;
+
+export interface Add extends Operation {
+	value: any;
```

Comment:
I'm not sure we can in a meaningful way. `Patch` allows for types to be specified, and it exports a `diff` method that will infer those types from its arguments. That's the primary way that the JSONPatch stuff will be used. 

But `Add` is adding a property at an arbitrary path in an object. If I wanted to create the
operation:
` { "op": "add", "path": "/a/b/c", "value": [ "foo", "bar" ] }`
I'm not sure how feasible it would be, or how much sense it would make, to try to enforce any kind of typing there.